src/components/Tables/Table.tsx

```diff
@@ -0,0 +1,150 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import DataTable, { TableColumn } from "react-data-table-component";
+import TableFilterBar, { FilterText } from "@/components/Tables/TableFilterBar";
+
+interface Row {
+    checkBoxKey: number;
+    [key: string]: string | number;
+}
+interface Headers {
+    [key: string]: string;
+}
+
+export interface Datum {
+    [key: string]: string | number | null;
+}
+
+interface Props {
+    data: Datum[];
+    headers: Headers;
+}
+
+const itemIncludesFilterText = (headers: Headers, item: Datum, filterText: FilterText): boolean => {
+    for (const key of Object.keys(headers)) {
+        if (
+            !(item[key] ?? "")
+                .toString()
+                .toLowerCase()
+                .includes((filterText[key] ?? "").toLowerCase())
+        ) {
+            return false;
+        }
+    }
+    return true;
+};
+
+const filteredNoNullItems = (headers: Headers, data: Datum[], filterText: FilterText): Row[] => {
+    let row = 0;
+    return data.reduce((filteredNoNullList: Row[], item: Datum, currentIndex: number) => {
+        if (itemIncludesFilterText(headers, item, filterText)) {
+            const noNullItem: Row = { checkBoxKey: currentIndex };
+            for (const key of Object.keys(headers)) {
+                noNullItem[key] = item[key] ?? "";
+            }
+            noNullItem.id = row;
+            filteredNoNullList.push(noNullItem);
+            row++;
+        }
+        return filteredNoNullList;
+    }, []);
+};
+
+const Table: React.FC<Props> = (props) => {
+    const [filterText, setFilterText] = useState<FilterText>({});
+
+    const [selectCheckBox, setSelectCheckBox] = useState(
+        new Array<boolean>(props.data.length).fill(false)
+    );
```

Comment:
The current naming of `selectCheckBox` implies that this is a single value, although it is actually an array of the check boxes for each row in the table.

Perhaps to `rowSelectBoxes` or `selectRowBoxes`? You'll need to decide what you think is best.

If you do decide to rename, then you'll need to also rename the other related variables